apps/api_web/test/api_web/params_test.exs

```diff
@@ -54,17 +54,46 @@ defmodule ApiWeb.ParamsTest do
         }
       }
 
-      {:ok, %{params: params}}
+      {:ok, %{params: params, conn: conn}}
     end
 
-    test "filters items from 2nd arg", %{params: params} do
-      assert Params.filter_params(params, ["route", "trip", "stop"]) ==
+    test "filters items from 2nd arg", %{params: params, conn: conn} do
+      assert Params.filter_params(params, ["route", "trip", "stop"], conn) ==
                {:ok,
                 %{
                   "route" => "1,2,3",
                   "trip" => "1,2,3",
                   "stop" => "4,5,6"
                 }}
     end
+
+    test "returns error in case of unsupported filter", %{params: params, conn: conn} do
+      assert Params.filter_params(params, ["route"], conn) == {:error, :bad_filter, ~w(stop trip)}
+
```

Comment:
These should probably be two separate tests.

Reply:
Fixed